### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by RashidMenhas**
## Followed Project requirements 💯 

✔✔✔ No Linter Error
✔✔✔ Use Correct GetFlow

- Connect an existing API via Redux and use the data to display something in a UI.
- Refactor my code to use the Bookstore API to add and remove books.
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch a list of books.
- Dispatch another action (add an action type) that will pass the API response to the reducer and update the state.
- Refactor add book and remove book features to persist changes in the server.
- Change the existing Action Creators, so they return functions.
- Make changes by using API.
- Dispatch an action (action type that existed before refactor) that will update the state accordingly.
- Add book and remove book should work in the same way after the refactor.

### CODE REVIEW

File: src/components/AllJS/books.js [hunk 1]
```diff
@@ -22,8 +23,8 @@ const Books = () => {
     if (titlecheck.value === '' || authorcheck.value === '') {
       alert('Please fill the field');
     } else {
+      api.addNewBook(book);
```
Comment (danmainah) [inline src/components/AllJS/books.js]: - [ ] @RashidMenhas great work on your component ,however this method is suppose to be in the reducer similar to the delete and getbooks methods.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -37,8 +16,21 @@ export const bookReducer = (state = initialdata, action) => {
...
+
+export const removeBook = (bookId) => async (dispatch) => {
+  api.deleteBook(bookId);
+  dispatch(REMOVE_BOOK(bookId));
+};
+
+export const getBooks = () => async (dispatch) => {
+  const books = await api.fetchBooks();
+  dispatch(GET_BOOKS(books));
+};
```
Comment (danmainah) [inline src/redux/books/books.js]: - [ ] @RashidMenhas great work on your reducer however it's a project requirement to use `createAsyncThunk`, kindly ensure your reducer users it.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @RashidMenhas , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 2 above)
Comment (danmainah) [reply src/redux/books/books.js]: @RashidMenhas as previously requested kindly use `createAsyncThunk` its a project requirement.
``example using it in a code``
```
export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },

```

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @RashidMenhas , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hi @RashidMenhas , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Required Changes ♻️
 - [ ] As requested by the previous reviewers, https://github.com/RashidMenhas/books-store/pull/4#discussion_r1032677735 kindly use createAsyncThunk as it's a project requirement. 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,32 +1,11 @@
...
-  },
-  {
-    id: '4',
-    title: ' Algorithm ',
-    author: 'Saboor',
-  },
-];
-
-export const bookReducer = (state = initialdata, action) => {
+export const bookReducer = (state = [], action) => {
```
Comment (KayLemba) [inline src/redux/books/books.js]: Hello  @RashidMenhas as previously requested kindly use createAsyncThunk as it is  a project requirement.
 Follow the example given below:

export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },

**Review by KayLemba — CHANGES_REQUESTED** [review]
Hi @RashidMenhas  👋
Good job so far working on the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight 🎉
No linter errors. ✔️
GitHub flow is followed. ✔️

Required Changes ♻️
Check the comments under the review in the inline comments.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**